glommio/src/sync/semaphore.rs

```diff
@@ -447,13 +520,21 @@ impl Semaphore {
     }
 }
 
+impl Drop for Semaphore {
+    fn drop(&mut self) {
+        assert!(self.state.borrow().waiters_list.is_empty());
+    }
+}
+
 #[cfg(test)]
 mod test {
     use super::*;
     use crate::timer::Timer;
     use crate::{enclose, Local, LocalExecutor};
 
+    use futures::future::join3;
```

Comment:
I am trying to avoid adding `futures` dependencies, and standardize on `futures_lite` (although there is no need to rush to remove the existing ones).

Worst case if you don't find anything in `futures_lite` you can just await them all separately. Better than increasing the dependency on `futures`

Reply:
Fixed